### ISSUE TITLE
Programmer Error present in Console upon launching Editor

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#6408]
**Describe the bug**
Launching the Editor with an active project having the Prefab system enabled, results in the following Programmer Error being recorded in the Console:

```
[Error] (Serialize) - Programmer error:
Classes deriving from AZ::Component are not declaring their base class to SerializeContext.
This will cause unexpected behavior such as components shifting around, or duplicating themselves.
Affected components:
- ProceduralPrefabSystemComponent {81211818-088A-49E6-894B-7A11764106B1}
Reflection code should look something like this:
serializeContext->Class<MyComponent, AZ::Component, ... (other base classes, if any) ...>()
Make sure the Reflect function is called for all base classes as well.
```
Please note that no level does not have to be loaded for the error to appear.

**Prerequisites**

- Prefabs are enabled.

**Steps to reproduce**
1. Launch the Editor.
2. Examine the Console log.

**Expected behavior**
Console log is free of Errors upon launching the Editor.

**Actual behavior**
Console log displays the Programmer Error.

**Screenshots/Video**

![ProgrammerError](https://user-images.githubusercontent.com/86953659/146014930-572ea8aa-7da5-4ec3-afbb-7e05660ed03e.PNG)

**Found in Branch**
Development (d951e1f)

**Desktop/Device**
Device: PC
OS: Windows
Version: 10
CPU: Intel i5-9600K
GPU: RTX 2070
Memory: 16GB

**Comment by jjjoness** [comment]
@LB-KacperKapusta This appears to have been fixed on the 12/16/2021 by amzn-mike. This no longer occurs.

**Comment by LB-KacperKapusta** [comment]
This has been verified as fixed on Development (5e8c1aa) branch. Closing the issue.
